### PR TITLE
Add a public NoError typealias so that importing Result is not necessary to use NoError

### PR DESCRIPTION
**Pull request by JaviSoto**

### CODE REVIEW

**Comment by JaviSoto** [comment]
This seems to compile for me

File: ReactiveCocoa/Swift/Property.swift [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import Foundation
 import enum Result.NoError
 
+public typealias NoError = Result.NoError
```
Comment (NachoSoto) [inline ReactiveCocoa/Swift/Property.swift]: Maybe add a comment about what this accomplishes?

**Comment by NachoSoto** [comment]
👍

**Comment by JaviSoto** [comment]
Confirmed it works via Carthage. If I cmd+click on a `NoError` symbol in the app, it takes me to the generated header of RAC with this:

screen shot 2016-02-04 at 12 27 56 pm

Do we want to make a new release with this?

**Comment by JaviSoto** [comment]
Hm, maybe hold off on this, I just saw a linker error

**Comment by JaviSoto** [comment]
Swift doesn't like this I'm afraid :(

```Undefined symbols for architecture x86_64:
  "direct type metadata for Result.NoError", referenced from:
      static ModuleX.TypeY.(methodName in _D91856F41B9960ECDC9FE90415B208AA) (ModuleX.TypeY.Type)() -> () in File.o
      static ModuleX.TypeY.((methodName in _D91856F41B9960ECDC9FE90415B208AA) (ModuleX.TypeY.Type) -> () -> ()).(closure #1) in ErrorReporting.o
      static ModuleX.TypeY.((methodName in _D91856F41B9960ECDC9FE90415B208AA) (ModuleX.TypeY.Type) -> () -> ()).(closure #1).(closure #1) in File.o
      type metadata accessor for ReactiveCocoa.SignalProducer<FabricAPI.TypeY?, Result.NoError> in File.o
      type metadata accessor for ReactiveCocoa.SignalProducer<FabricAPI.TypeY?, Result.NoError> in File.o
      ModuleX.TypeY.init (Instant.ModuleX.TypeY.Type)(parameter : ModuleX.TypeY) -> ModuleX.TypeY in File.o
      ModuleX.TypeY.method (ModuleX.TypeY)(ModuleX.TypeY) -> () in File.o
      ...
  "protocol witness table for Result.NoError : Swift.ErrorType in Result", referenced from:
      static ModuleX.TypeY.(methodName in _D91856F41B9960ECDC9FE90415B208AA) (ModuleX.TypeY.Type)() -> () in File.o
      static Instant.ErrorReporting.((attachUserInformationToCrashReports in _D91856F41B9960ECDC9FE90415B208AA) (Instant.ErrorReporting.Type) -> () -> ()).(closure #1) in File.o
      static Instant.ErrorReporting.((attachUserInformationToCrashReports in _D91856F41B9960ECDC9FE90415B208AA) (Instant.ErrorReporting.Type) -> () -> ()).(closure #1).(closure #1) in ErrorReporting.o
      type metadata accessor for ReactiveCocoa.SignalProducer<FabricAPI.AuthenticatedFabricAPI?, Result.NoError> in ErrorReporting.o
      type metadata accessor for ReactiveCocoa.SignalProducer<FabricAPI.AuthenticatedUser?, Result.NoError> in ErrorReporting.o
      Instant.NotificationsMutingViewModel.init (Instant.NotificationsMutingViewModel.Type)(fabricAPI : FabricAPI.AuthenticatedFabricAPI) -> Instant.NotificationsMutingViewModel in NotificationsMutingViewModel.o
      Instant.NotificationsMutingViewModel.setMuteSettingTimePeriod (Instant.NotificationsMutingViewModel)(Instant.MuteSettingTimePeriod) -> () in NotificationsMutingViewModel.o
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by JaviSoto** [comment]
I'm afraid Swift doesn't like this...

```
Undefined symbols for architecture x86_64:
  "direct type metadata for Result.NoError", referenced from:
      static ModuleX.TypeY.(methodName in _D91856F41B9960ECDC9FE90415B208AA) (ModuleX.TypeY.Type)() -> () in File.o
      static ModuleX.TypeY.((methodName in _D91856F41B9960ECDC9FE90415B208AA) (ModuleX.TypeY.Type) -> () -> ()).(closure #1) in File.o
      static ModuleX.TypeY.((methodName in _D91856F41B9960ECDC9FE90415B208AA) (ModuleX.TypeY.Type) -> () -> ()).(closure #1).(closure #1) in File.o
      type metadata accessor for ReactiveCocoa.SignalProducer<ModuleX.TypeY?, Result.NoError> in File.o
      type metadata accessor for ReactiveCocoa.SignalProducer<ModuleX.TypeY?, Result.NoError> in File.o
      ModuleX.TypeY.init (ModuleX.TypeY.Type)(parameter : ModuleX.TypeY) -> ModuleX.TypeY in File.o
      ModuleX.TypeY.method (ModuleX.TypeY)(ModuleX.TypeY) -> () in File.o
      ...
  "protocol witness table for Result.NoError : Swift.ErrorType in Result", referenced from:
      static ModuleX.TypeY.(methodName in _D91856F41B9960ECDC9FE90415B208AA) (ModuleX.TypeY.Type)() -> () in File.o
      static ModuleX.TypeY.((methodName in _D91856F41B9960ECDC9FE90415B208AA) (ModuleX.TypeY.Type) -> () -> ()).(closure #1) in File.o
      static ModuleX.TypeY.((methodName in _D91856F41B9960ECDC9FE90415B208AA) (ModuleX.TypeY.Type) -> () -> ()).(closure #1).(closure #1) in File.o
      type metadata accessor for ReactiveCocoa.SignalProducer<ModuleX.TypeY?, Result.NoError> in File.o
      type metadata accessor for ReactiveCocoa.SignalProducer<ModuleX.TypeY?, Result.NoError> in File.o
      ModuleX.TypeY.init (ModuleX.TypeY.Type)(parameter : ModuleX.TypeY) -> ModuleX.TypeY in File.o
      ModuleX.TypeY.setMuteSettingTimePeriod (ModuleX.TypeY)(ModuleX.TypeY) -> () in File.o
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by vasarhelyia** [comment]
:cry:

**Comment by NachoSoto** [comment]
Nothing we can do then. I'll update the release notes and docs to make it clear that you have to import `ReactiveCocoa` and `Result`.

**Pull request closed without merging**